cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -394,14 +394,53 @@ func GetCoreDNSInfo(client clientset.Interface) (*v1.ConfigMap, string, string,
 	return coreDNSConfigMap, corefile, currentCoreDNSversion, nil
 }
 
+type objectForUpdateVolumeStrategicMergePatch struct {
+	Spec objectSpecForMergePatch `json:"spec"`
+}
+
+type objectSpecForMergePatch struct {
+	Volumes []objectVolumeForPatch `json:"volumes"`
+}
+
+type objectVolumeForPatch struct {
+	Name      string            `json:"name"`
+	Configmap configmapForPatch `json:"configmap"`
+}
+
+type configmapForPatch struct {
+	Name  string              `json:"name"`
+	Items []map[string]string `json:"items"`
+}
+
 func patchCoreDNSDeployment(client clientset.Interface, coreDNSCorefileName string) error {
 	dnsDeployment, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Get(kubeadmconstants.CoreDNSDeploymentName, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
-	patch := fmt.Sprintf(`{"spec":{"template":{"spec":{"volumes":[{"name": "config-volume", "configMap":{"name": "coredns", "items":[{"key": "%s", "path": "%s"}]}}]}}}}`, coreDNSCorefileName, coreDNSCorefileName)
```

Comment:
If you want to prevent possible JSON injection here, why don't replace `"%s"` with `%q` instead?
In addition to that, `patchCoreDNSDeployment` is called in a single spot with a constant string value.

Reply:
I think, this treatment could unify the encoding method to ensure encoded values are validated before use.